torchgen/native_function_generation.py

```diff
@@ -263,11 +265,31 @@ def generate_function(
             # which NativeFunction objects did not come directly from native_functions.yaml.
             tags=set(["generated"]),
             namespace=f.namespace,
+            symint_args=[],
         ),
         backend_metadata,
     )
 
 
+def ints_to_symints_nativefunction(f):
+    assert "symint_ver_needed" in f.tags
+    new_post_self_positional = []
+    for arg in f.func.arguments.post_self_positional:
+        if arg.name in f.symint_args:
+            new_post_self_positional.append(arg.replace_with_base_type(BaseType(BaseTy.int), BaseType(BaseTy.SymInt)))
+        else:
+            new_post_self_positional.append(arg)
+    
+    
+    new_schema = f.func.replace_with_base_type(BaseType(BaseTy.int), BaseType(BaseTy.SymInt))
```

Comment:
This function needs to get tightened up. Why do you do a replacement on schema, but then manually overwrite the arguments afterwards?